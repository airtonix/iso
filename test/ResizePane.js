const React = require('react')
const { create: render } = require('react-test-renderer')
const {
  Simulate,
  renderIntoDocument
} = require('react-dom/test-utils')
const ResizePane = require('../src/ResizePane').default

describe('ResizePane', () => {
  test('renders', () => {
    const json = render(
      React.createElement(ResizePane, {})
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('has initial state', () => {
    const instance = render(
      React.createElement(ResizePane, {
        width: 448
      })
    ).getInstance()
    expect(instance.state.active).toBe(false)
    expect(instance.state.mouse).toBe(null)
    expect(instance.state.width).toBe(448)
  })

  test('handles mousedown event', () => {
    const pane = renderIntoDocument(
      React.createElement(ResizePane, {})
    )
    Simulate.mouseDown(pane.root, {
      clientX: 4,
      clientY: 4,
    })
    expect(pane.state.mouse).toEqual({
      x: 4,
      y: 4
    })
  })

  test.skip('handles mousemove event', () => {})
  test.skip('handles mouseup event', () => {})
  test.skip('handles click event', () => {})
})
