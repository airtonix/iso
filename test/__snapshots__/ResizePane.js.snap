// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResizePane renders 1`] = `
<div
  onClick={[Function]}
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
  style={
    Object {
      "cursor": "ew-resize",
      "userSelect": "none",
      "width": undefined,
    }
  }
/>
`;
